module/VuFindSearch/tests/unit-tests/src/VuFindTest/Backend/Solr/ConnectorTest.php

```diff
@@ -140,6 +140,41 @@ public function testSaveThrowsUnknownFormat()
         $conn->write($document, 'unknown', 'update');
     }
 
+    /**
+     * Test caching.
+     *
+     * @return void
+     */
+    public function testCaching()
+    {
+        $conn = $this->createConnector('single-record');
+
+        list(, $expectedBody) = explode("\n\n", $this->response);
+
+        $cache = $this->createMock(\Laminas\Cache\Storage\StorageInterface::class);
+        $cache->expects($this->exactly(3))
+            ->method('getItem')
+            ->willReturnOnConsecutiveCalls(null, $expectedBody, 'foo');
```

Comment:
You could add a `with` here to indicate that all three calls have the same input parameter; that might help clarify things slightly.

Reply:
`with` added. It's still just a string constraint since I think hard-coding the actual key just makes it more complicated to maintain without obvious benefit.